lib/avromatic/model/custom_type.rb

```diff
@@ -0,0 +1,55 @@
+require 'avromatic/model/null_custom_type'
+
+module Avromatic
+  module Model
+
+    # Instances of this class contains the configuration for custom handling of
+    # a named type (record, enum, fixed).
+    class CustomType
+
+      attr_accessor :to_avro, :from_avro, :value_class
+
+      def initialize(value_class)
+        @value_class = value_class
+      end
+
+      # A coercer method is used when assigning to the model. It is used both when
+      # deserializing a model instance from Avro and when directly instantiating
+      # an instance. The coecer method must accept a single argument and return
+      # the value to store in the model for the attribute.
+      def coercer
```

Comment:
I don't like the `avro_decoder` and `avro_encoder` names since these methods aren't dealing with Avro encoded values. It's "decoding" after the Avro decode, and encoding prior to the Avro encode.

If we want something symmetric how about `deserializer` and `serializer`? 

The only other idea I have is something around conversion: `model_converter` and `avro_converter`, or `model_coercer` and `avro_coercer` (where the methods are named for the destination).